### DISCUSSION TITLE
myst and jupyter-nbconvert --to slides

### DISCUSSION THREAD

**Discussion by paugier** [executablebooks/meta#772] [Q&A]
I write reveal.js presentations with jupyter-lab and `jupyter-nbconvert --to slides`.

I would love to be able to use myst syntax for these presentations but I'm not able to use it with jupyter-nbconvert. Could it work?

For some cases in these presentations, I need to write simple html. I see few features that are very important to produce nice presentations. It should be easy to:

- organize in columns (with things like `<div style="float:left;width:48%">`)
- add texts and images at precise places on a image
- tweak a bit the css for a particular object

The problem with doing this with html is that (i) the syntax is not as nice as myst and (ii) I can't directly mix html and markdown (for example add a markdown list in a div used to have a column).

I don't know well what can be done now with myst, but it seems to me that the language is powerful enough to describe such things.

So I feel that the jupyter and myst technologies could be used together to produce a very good tool to write html presentations. But I can't find in the documentations of the different projects what is doable now and I don't know if this could be a goal for these projects to be used for this purpose (writing nice web slides).

**Comment by paugier** [comment]
Note that I tried something simple which does not work.

```
from IPython.display import display as display_base, HTML
from myst_parser.main import to_html

def display_myst(code):
    return display_base(HTML(to_html(code)))
```

Then one could produce html code from myst code in a cell with for example
````
display_myst("""
```{admonition} [jupyter-book](https://jupyterbook.org) is now a great project!

blablabla...
```
""")
````

This produces the right html but I guess one would need to somehow inject myst css in the html file of the presentation.

**Reply by paugier** [reply]
I opened [a related issue](https://github.com/jupyter/nbconvert/issues/1806) in nbconvert issue tracker.

**Comment by chrisjsewell** [comment]
Heya, ping me if I forget to respond fully (as a bit short on time now 😅), but one thing I was interested in recently was: https://marpit.marp.app/. I think this may be a good "syntax" basis to crib from.
This would be for "standards Markdown", then you could just extend this to the concept of a Notebook.

I guess the main issue is integration with docutils/sphinx. There is no real *slide* output formats available for those, so if you wanted them I guess you would need to write your own "output translator" (probably subclassing from the HTML one)

**Reply by chrisjsewell** [reply]
there is also this open issue: https://github.com/executablebooks/jupyter-book/issues/792#issuecomment-905798366

**Reply by mmcky** [reply]
We have been playing around with the idea of `slides` at QuantEcon for a while now. We have basically concluded the same as you @chrisjsewell. I think the "best" combination would be a `SlideWriter` + some directives such as:

````markdown
```{slide}
<content>
```
````

We've been thinking this through given it would be nice for a lot of projects to have accompanying slides for lectures and education material. But it is hard to "summarise" information automatically so an embedded slide directive near the other lecture based content was one idea we cam up with with such that it is ignored by `html` writer and written using the `slide` writer to embed as a slides version of the page.

**Reply by paugier** [reply]
Hi @chrisjsewell 

After some thoughts, I really think a .myst.md to reveal.js converter would be very useful. I created a quasi empty repository with some notes https://github.com/paugier/myst-pres.

I can try to work a bit on this project.

Feedback and advises on the implementation would be very useful.
